buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlRestCompatTestPlugin.java

```diff
@@ -151,8 +155,13 @@ public void apply(Project project) {
         // setup IDE
         GradleUtils.setupIdeForTestSourceSet(project, yamlCompatTestSourceSet);
 
-        // wire this task into check
-        project.getTasks().named(JavaBasePlugin.CHECK_TASK_NAME).configure(check -> check.dependsOn(yamlRestCompatTestTask));
+        // wire this task into the custom check task
+        project.getTasks().named(CheckRestCompatPlugin.CHECK_TASK_NAME).configure(check -> check.dependsOn(yamlRestCompatTestTask));
+    }
+
+    private boolean isEnabled(Project project) {
+        Object bwcEnabled = project.getExtensions().getExtraProperties().getProperties().get("bwc_tests_enabled");
```

Comment:
@breskeby is this method of accessing extra properties kosher with the configuration cache?

Reply:
I think what breaks configuration cache here would be accessing project. I think we need to change how `bwc_tests_enabled` is accessed in general and not just here. So I'm fine to keep this for now here. We will revisit when we rework the general way of accessing build properties in a configuration cache friendly manner

Reply:
Agreed. I think that should be a property on `BuildParams` instead.

Reply:
Agreed. We should just move it to `gradle.properties` really instead of using extra properties for this.